src/nwb_conversion_tools/tools/data_transfers.py

```diff
@@ -84,3 +89,62 @@ def estimate_s3_conversion_cost(
     cost_gb_m = 0.08 / 1e3  # $0.08 / GB Month
     cost_mb_s = cost_gb_m / (1e3 * 2.628e6)  # assuming 30 day month; unsure how amazon weights shorter months?
     return cost_mb_s * total_mb_s
+
+
+def automatic_dandi_upload(
+    nwb_folder_path: FolderPathType,
+    dandiset_id: str,
+    version: Optional[str] = None,
+    staging: bool = False,
+    api_token: Optional[str] = None,
+):
+    """
+    Fully automated upload of NWBFiles to a DANDISet.
+
+    Parameters
+    ----------
+    nwb_folder_path : FolderPathType
+        Folder containing the NWBFiles to be uploaded.
+    dandiset_id : str
+        Six-digit string identifier for the DANDISet the NWBFiles will be uploaded to.
+    version : str, optional
+        "draft" or "version".
+        The default is "draft".
+    staging : bool, optional
+        Is the DANDISet hosted on the staging server? This is mostly for testing purposes.
+        The default is False.
+    api_token : str
+        Your API token for your DANDI account - DO NOT STORE THIS IN ANY CODE ON GITHUB.
+        Use environment variables for CI, or interactivity for personal usage.
+        Or store personal conversion scripts containing the token outside of GitHub.
+    """
+    version = "draft" if version is None else version
+    if api_token is None:
+        api_token = input("Please enter your DANDI API token: ")
+    url_base = "https://gui-staging.dandiarchive.org" if staging else "https://dandiarchive.org"
+    dandiset_url = f"{url_base}/dandiset/{dandiset_id}/{version}"
+
+    os.chdir(nwb_folder_path.parent)
+    validate_return = os.popen(f"dandi validate {nwb_folder_path.name}").read().strip()
+    assert re.fullmatch(
+        pattern=r"^Summary: No validation errors among \d+ file\(s\)$", string=validate_return
+    ), "DANDI validation failed!"
+
+    download_return = os.popen(f"dandi download {dandiset_url}").read()
```

Comment:
are you downloading data within the NWB folder? I don't know if that's the best idea. It requires write access to the NWB folder, which you may not have, and in general I'd prefer to keep the staged dandiset in a different spot from the original NWB files.

Reply:
> are you downloading data within the NWB folder?

Ah, yes technically if the dandiset has previous uploads this will grab those unintentially. I'll add `--download dandiset.yaml`, we just need the shell folder for upload setup.

Reply:
Also added dandiset folder specification so user can control where it gets downloaded in the event the nwb folder source doesn't have write access